assets/blocks/digital-wallets/hooks.js

```diff
@@ -0,0 +1,347 @@
+	}, [payments, needsShipping]);
+
+	return paymentRequest;
+}
+
+/**
+ * Registers event handler on `shippingcontactchanged`
+ *
+ * @param {Object} paymentRequest The payment request object.
+ * @return {Function} Function to remove the listener on unmount.
+ */
+export function useShippingContactChangeHandler(paymentRequest) {
+	useEffect(() => {
+		paymentRequest?.addEventListener(
+			'shippingcontactchanged',
+			(shippingContact) => handleShippingAddressChanged(shippingContact)
+		);
+	}, [paymentRequest]);
+
+	return () => paymentRequest?.removeListener('shippingcontactchanged');
+}
+
+/**
+ * Registers event handler on `shippingoptionchanged`
+ *
+ * @param {Object} paymentRequest The payment request object.
+ * @return {Function} Function to remove the listener on unmount.
+ */
+export function useShippingOptionChangeHandler(paymentRequest) {
+	useEffect(() => {
+		paymentRequest?.addEventListener('shippingoptionchanged', (option) =>
+			handleShippingOptionChanged(option)
+		);
+	}, [paymentRequest]);
+
+	return () => paymentRequest?.removeListener('shippingoptionchanged');
+}
+
+/**
+ * Initializes Google Pay with the provided payments and paymentRequest objects.
+ *
+ * @param {Object} payments       The Square payments object used to create the payment request.
+ * @param {Object} paymentRequest The payment request object.
+ * @return {Array} Array containing the Google Pay instance and its reference.
+ */
+export function useGooglePay(payments, paymentRequest) {
+	const [googlePay, setGooglePay] = useState(null);
+	const googlePayRef = useRef(null);
+
+	useEffect(() => {
+		if (!(payments && paymentRequest)) {
+			return;
+		}
+
+		(async () => {
+			try {
+				await payments.googlePay(paymentRequest);
+				const __googlePay = await payments.googlePay(paymentRequest);
+				await __googlePay.attach(googlePayRef.current, {
+					buttonColor: getSquareServerData().googlePayColor,
+					buttonSizeMode: 'fill',
+					buttonType: 'long',
+				});
+
+				setGooglePay(__googlePay);
+			} catch (e) {}
+		})();
+	}, [payments, paymentRequest]);
+
+	return [googlePay, googlePayRef];
+}
+
+/**
+ * Initializes Apple Pay with the provided payments and paymentRequest objects.
+ *
+ * @param {Object} payments       The Square payments object used to create the payment request.
+ * @param {Object} paymentRequest The payment request object.
+ * @return {Array} Array containing the Apple Pay instance and its reference.
+ */
+export function useApplePay(payments, paymentRequest) {
+	const [applePay, setApplePay] = useState(null);
+	const applePayRef = useRef(null);
+
+	useEffect(() => {
+		if (!(payments && paymentRequest)) {
+			return;
+		}
+
+		(async () => {
+			try {
+				await payments.applePay(paymentRequest);
+				const __applePay = await payments.applePay(paymentRequest);
+
+				setApplePay(__applePay);
+			} catch (e) {}
+		})();
+	}, [payments, paymentRequest]);
+
+	useEffect(() => {
+		if (!applePayRef?.current) {
```

Comment:
Required with above suggestion to ensure the Apple Pay button is only shown in Safari.

```suggestion
		if (!applePayRef?.current || !applePay) {
```

Reply:
@peterwilsoncc I've added these suggestions.